### PR TITLE
Adding pylint to travis

### PR DESCRIPTION
**Pull request by yaacov**
**Description**

Adding pylint to travis.

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -17,6 +17,8 @@
 # limitations under the License.
 #
 
+""" build.py makefile helper """
```
Comment (jhernand) [inline build.py]: The `Makefile` is the helper, not this.

File: build.py [hunk 2]
```diff
@@ -109,6 +115,9 @@ def go_tool(*args):
 
 @cache
 def ensure_project_dir():
+    """
+    returns the project directory.
```
Comment (jhernand) [inline build.py]: Sentences should start with upper case.

File: build.py [hunk 3]
```diff
@@ -323,7 +332,12 @@ def fmt():
 
 
 def main():
-    # Create the top level command line parser:
+    """
+    Create the top level command line parser:
```
Comment (jhernand) [inline build.py]: This is not what the method does. If `pylint` really want this comment then write something that describes the method, not just the first line.

File: build.py [hunk 4]
```diff
@@ -368,10 +382,10 @@ def main():
 
     # Run the selected tool:
     code = 0
+    # pylint: disable=len-as-condition
     if len(sys.argv) > 0 and sys.argv[1] in DIRECT_TOOLS:
```
Comment (jhernand) [inline build.py]: I think you can change this to `if sys.argv and ...` instead of asking `pylint` to ignore.

Comment (yaacov) [reply build.py]: :+1:

Comment (cben) [reply build.py]: Actually this caught a real bug — argv being non-empty only guarantees sys.argv[0] but not sys.argv[1]:
```
$ ./build.py 
Traceback (most recent call last):
  File "./build.py", line 605, in <module>
    main()
  File "./build.py", line 590, in main
    if len(sys.argv) > 0 and sys.argv[1] in DIRECT_TOOLS:
IndexError: list index out of range
```

Sending a PR to fix...

**Pull request merged**